test/test_sparse.py

```diff
@@ -298,7 +299,8 @@ def fn(x):
         ])
         # we don't have to_dense for half types on CPU because it is implemented
         # with a slower add_ operation
-        for dtype in [torch.float16, torch.float32, torch.float64] if self.device != 'cpu' else [torch.float32, torch.float64]:
+        for dtype in [torch.float16, torch.float32, torch.float64, torch.cfloat, torch.cdouble] \
```

Comment:
We really should invest in updating the sparse tests to use the device generic test framework so we can use the @dtypes decorator.